app/components/FileUpload.jsx

```diff
@@ -9,27 +9,73 @@ const fileTypes = ["csv"]
 
 const FileUpload = () => {
   const [file, setFile] = useState(null);
+  const [dataToUpload, setDataToUpload] = useState(null);
+  const [extractedZoneCode, setExtractedZoneCode] = useState(null);
+  const [isCodeRecognized, setIsCodeRecognized] = useState(true);
+
   const handleChange = (file) => {
     Array.from(file)
     .filter((file) => file.type === "text/csv")
-    .forEach( async (file) => {
+    .forEach(async (file) => {
 
       const text = await file.text();
       
-      csvToJson(text, (err, data) => {
+      csvToJson(text, async (err, data) => {
         if (err) {
           console.log(err);
         } else {
-          console.log(data);
-          //TODO YS's algorithm will be implemented here
-          console.log(calculateHourlyAverages(data))
+          //Zone Code List : Additional list of zone codes will be added manually
+
+          const dataToStore = calculateHourlyAverages(data)
+          setDataToUpload(dataToStore);
 
+          //Extracting Zone Code from the data
+          const zoneCodeList = [
```

Comment:
Will update this logic later.

Still having a doubt about the const list in here.

Reply:
Seems hard coded - will not work if zone names are changed in the future